Resources/Prototypes/Recipes/Construction/Graphs/firelock.yml

```diff
@@ -0,0 +1,141 @@
+﻿- type: constructionGraph
+  id: Firelock
+  start: start
+  graph:
+    - node: start
+      edges:
+        - to: frame1
+          completed:
+            - !type:SnapToGrid { }
+          steps:
+            - material: Metal
+              amount: 3
+              doAfter: 1
+
+
+    - node: frame1
```

Comment:
Give this node an action that changes the sprite state to frame1. For reversing the construction.